namesys/routing.go

```diff
@@ -19,25 +23,74 @@ var log = logging.Logger("namesys")
 // routingResolver implements NSResolver for the main IPFS SFS-like naming
 type routingResolver struct {
 	routing routing.IpfsRouting
+
+	cache     *lru.Cache
+	cachelock sync.Mutex
+	cachelife time.Duration
 }
 
-// NewRoutingResolver constructs a name resolver using the IPFS Routing system
-// to implement SFS-like naming on top.
-func NewRoutingResolver(route routing.IpfsRouting) Resolver {
-	if route == nil {
-		panic("attempt to create resolver with nil routing system")
+func (r *routingResolver) cacheGet(name string) (path.Path, bool) {
+	r.cachelock.Lock()
+	ientry, ok := r.cache.Get(name)
```

Comment:
2nding @ion1 here. The locking isn't needed. 

https://github.com/hashicorp/golang-lru/blob/master/lru.go#L87